### DISCUSSION TITLE
How do I show the video live?

### DISCUSSION THREAD

**Discussion by shubham190398** [ultralytics/yolov5#9014] [Q&A] [answered]
Hello. I wanted to use this to get the video live in colab after detecting the classes in a source video rather than saving the output. What would be the best way to go about it?

**Comment by glenn-jocher** [comment, marked as answer]
@shubham190398 Colab doesn't support showing windows, but you can do this locally with `python detect.py --view-img`

**Reply by shubham190398** [reply]
Thanks a lot! So it would basically be `--view -img` along with whatever I am detecting for a video/image? Can I edit the window size here?

**Reply by glenn-jocher** [reply]
@shubham190398 yes that's correct. Window should be resizable.

**Reply by MilesJoseph** [reply]
Hey @glenn-jocher, when I run the following 

!python detect.py --source Salmon_Escapement.mp4 --weights runs/train/exp/weights/best.pt --conf .40 --save-txt --view-img

I get the following error..... 

 Fusing layers... 
Model summary: 213 layers, 7012822 parameters, 0 gradients, 15.8 GFLOPs
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/usr/local/lib/python3.7/dist-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.

Have you seen this? I can run it fine without the view img part. I understand google collab doesn't support but how can I view/verify on collab as the scored video is reduced to one second. See below. 

Another question I have though, is that when I do run it without that the --view-img it seems like the inference is only running for a certain portion of the video. For instance; 

Fusing layers... 
Model summary: 213 layers, 7012822 parameters, 0 gradients, 15.8 GFLOPs
video 1/1 (1/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 12.2ms
video 1/1 (2/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (3/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.3ms
video 1/1 (4/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (5/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (6/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.4ms
video 1/1 (7/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (8/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (9/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.3ms
video 1/1 (10/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (11/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (12/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 11.6ms
video 1/1 (13/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.3ms
video 1/1 (14/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (15/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (16/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.3ms
video 1/1 (17/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.3ms
video 1/1 (18/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (19/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (20/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.2ms
video 1/1 (21/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (22/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.3ms
video 1/1 (23/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (24/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (25/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.5ms
video 1/1 (26/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (27/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (28/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.3ms
video 1/1 (29/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 1 Salmon, 10.3ms
video 1/1 (30/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
video 1/1 (31/26261) /content/yolov5/Salmon_Escapement.mp4: 480x640 (no detections), 10.2ms
Speed: 0.6ms pre-process, 10.4ms inference, 0.6ms NMS per image at shape (1, 3, 640, 640)
Results saved to runs/detect/exp4

It stops after 31 frames.... Can you give me insight to this? And, last question, when I look at the run, the video is reduced to one second....Is --view-img the only way to view the image in actual time? 

Anyways, I know that's a lot of questions, but I appreciate the time. Final bonus question, is --save-txt the only way to get analytic data from the data? Is there any dataframe output?

**Reply by glenn-jocher** [reply]
@MilesJoseph 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃
